### PR TITLE
Upgrade samples to 0.10.0

### PR DESCRIPTION
**Pull request by rorbech**

### CODE REVIEW

File: Intro/shared/src/commonMain/kotlin/io/realm/example/kmmsample/Calculator.kt [hunk 1]
```diff
@@ -30,7 +31,7 @@ class Calculator private constructor() {
             return a + b
         }
 
-        fun listen(): Flow<List<Expression>> {
+        fun listen(): Flow<RealmResults<Expression>> {
```
Comment (cmelchior) [inline Intro/shared/src/commonMain/kotlin/io/realm/example/kmmsample/Calculator.kt:34]: Any reason for this change? I assumed we used `List` to not depend too much on Realm specific things in the UI layer?

File: Intro/shared/src/commonMain/kotlin/io/realm/example/kmmsample/ExpressionRepository.kt [hunk 2]
```diff
@@ -39,7 +41,7 @@ class ExpressionRepository {
         return realm.query<Expression>().find()
     }
 
-    fun observeChanges(): Flow<List<Expression>> {
-        return realm.query<Expression>().asFlow()
+    fun observeChanges(): Flow<RealmResults<Expression>> {
```
Comment (cmelchior) [inline Intro/shared/src/commonMain/kotlin/io/realm/example/kmmsample/ExpressionRepository.kt:44]: Same

**Review by cmelchior — APPROVED** [review]
Nice 👍

File: MultiplatformDemo/shared/src/commonMain/kotlin/io/realm/kotlin/demo/model/CounterRepository.kt [hunk 3]
```diff
@@ -71,10 +73,11 @@ class CounterRepository {
      */
     fun observeCounter(): Flow<Long> {
         return realm.query<Counter>("id = 'primary'").asFlow()
```
Comment (nhachicha) [inline MultiplatformDemo/shared/src/commonMain/kotlin/io/realm/kotlin/demo/model/CounterRepository.kt:75]: why not using a SingleQuery?
```Kotlin
return realm.query<Counter>("id = 'primary'").first().asFlow()
            .map { it: SingleQueryChange<Counter> ->
                it.obj?.operations!!.fold(0L,) { sum, el -> sum + el }
            }
```

**Review by nhachicha — COMMENTED** [review]
Nice refactor with the version handling, minor comments

File: JVMConsole/build.gradle.kts [hunk 4]
```diff
@@ -2,18 +2,29 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
...
+        classpath("io.realm.kotlin:gradle-plugin:0.10.0")
+    }
+}
+rootProject.extra["realmVersion"] = "0.10.0"
+
+apply(plugin = "io.realm.kotlin")
+
 group = "io.realm.example"
-version = "1.0"
+version = "0.10.0"
```
Comment (nhachicha) [inline JVMConsole/build.gradle.kts:18]: don't forget to update its Readme to use 
```
java -jar build/libs/JVM_Console-0.10.0.jar
```

File: Bookshelf/shared/build.gradle.kts [hunk 5]
```diff
@@ -30,7 +30,6 @@ kotlin {
     val ktorVersion = "1.6.1"
     val serializationVersion = "1.2.1"
     val coroutinesVersion = "1.5.2-native-mt"
```
Comment (nhachicha) [inline Bookshelf/shared/build.gradle.kts:32]: any reason why this is still set to `1.5.x` it could send the wrong message since we're requiring `1.6.x`

**Review by clementetb — APPROVED** [review]
💎

File: MultiplatformDemo/shared/src/commonMain/kotlin/io/realm/kotlin/demo/model/CounterRepository.kt (hunk 3 above)
Comment (rorbech) [reply MultiplatformDemo/shared/src/commonMain/kotlin/io/realm/kotlin/demo/model/CounterRepository.kt]: Yeah, pretty cool simplification 🎉

File: JVMConsole/build.gradle.kts (hunk 4 above)
Comment (rorbech) [reply JVMConsole/build.gradle.kts:18]: Good catch!

File: Bookshelf/shared/build.gradle.kts (hunk 5 above)
Comment (rorbech) [reply Bookshelf/shared/build.gradle.kts]: I just forgot one ... shuffling through five Intellij and five Xcode instances 🤯 🤪

**Pull request merged**